modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIMANAGER4373BrokenAPIInStoreTestCase.java

```diff
@@ -0,0 +1,162 @@
+            apiStoreRestClient = new APIStoreRestClient(storeURLHttp);
+            contextUsername = publisherContext.getContextTenant().getContextUser().getUserName();
+            contextUserPassword = publisherContext.getContextTenant().getContextUser().getPassword();
+            userManagementClient = new UserManagementClient(publisherContext.getContextUrls().getBackEndUrl(),
+                    contextUsername, contextUserPassword);
+
+            userManagementClient
+                    .addRole(FIRST_ROLE, new String[] {}, new String[] { PERMISSION_LOGIN, PERMISSION_API_SUBSCRIBE });
+            userManagementClient
+                    .addRole(SECOND_ROLE, new String[] {}, new String[] { PERMISSION_LOGIN, PERMISSION_API_SUBSCRIBE });
+            userManagementClient.addUser(FIRST_USER, USER_PASSWORD, new String[] { FIRST_ROLE }, FIRST_USER);
+        } catch (APIManagerIntegrationTestException e) {
+            assertTrue(false, "Error occurred while initializing testcase.");
+        } catch (RemoteException e) {
+            assertTrue(false, "Error occurred while adding new users.");
+        } catch (UserAdminUserAdminException e) {
+            assertTrue(false, "Error occurred while adding new users.");
+        } catch (XPathExpressionException e) {
+            assertTrue(false, "Error occurred while retrieving context.");
+        }
+    }
+
+    @Test(groups = "wso2.am", description = "Test effect of changing the role of subscribed API")
+    public void testAPIRoleChangeEffectInStore() throws Exception {
+        // create two apis
+        apiPublisher.login(contextUsername, contextUserPassword);
+        APIRequest brokenApiRequest = new APIRequest(BROKEN_API, BROKEN_API, new URL(EP_URL));
+        brokenApiRequest.setVersion(API_VERSION);
+        brokenApiRequest.setProvider(contextUsername);
+        brokenApiRequest.setVisibility("restricted");
+        brokenApiRequest.setRoles(FIRST_ROLE);
+        apiPublisher.addAPI(brokenApiRequest);
+        APILifeCycleStateRequest updateRequest = new APILifeCycleStateRequest(BROKEN_API, contextUsername,
+                APILifeCycleState.PUBLISHED);
+        apiPublisher.changeAPILifeCycleStatus(updateRequest);
+
+        APIRequest healthyApiRequest = new APIRequest(HEALTHY_API, HEALTHY_API, new URL(EP_URL));
+        healthyApiRequest.setVersion(API_VERSION);
+        healthyApiRequest.setProvider(contextUsername);
+        healthyApiRequest.setVisibility("restricted");
+        healthyApiRequest.setRoles(FIRST_ROLE);
+        apiPublisher.addAPI(healthyApiRequest);
+        updateRequest = new APILifeCycleStateRequest(HEALTHY_API, contextUsername, APILifeCycleState.PUBLISHED);
+        apiPublisher.changeAPILifeCycleStatus(updateRequest);
+
+        // subscribe both apis
+        apiStoreRestClient.login(FIRST_USER, USER_PASSWORD);
+
+        apiStoreRestClient.addApplication(APP_NAME, "Unlimited", "", "");
```

Comment:
Better use constant for tier name. APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED can we used here.